File: autumn/human-readable-number/js/log2.js
function numberFormat(n) {
   let n2 = Math.log2(n) / 10;
   let n3 = Math.floor(n2);
   //return (n / 1000 ** n3).toFixed(3) + ['', ' k', ' M', ' G'][n3];
   return (n / 1000 ** n3) + ['', ' k', ' M', ' G'][n3];
}

let s = numberFormat(5678901234);
console.log(s);

File: autumn/human-readable-number/py/log2.py
import math

def f(n2):
   n = math.log2(n2) / 10
   n = math.floor(n)
   return '%.3f' % (n2 / 1024 ** n) + ('', ' k', ' M', ' G')[n]

s = f(1264)
print(s)
